packages/serve/src/createConfig.ts

```diff
@@ -18,11 +22,17 @@ export default function createConfig(args): devServerOptionsType {
         // clientLogging is not a valid devServer option
         delete options.clientLogging;
     }
-
-    if (options.hotOnly) {
-        options.hot = 'only';
-        // hotOnly is not a valid devServer option
-        delete options.hotOnly;
+    if (isDevServer4) {
+        if (options.hotOnly) {
+            options.hot = 'only';
+            // hotOnly is not a valid devServer option
+            delete options.hotOnly;
+        }
+    } else {
+        // only apply hotOnly when both are supplied
+        if (options.hot && options.hotOnly) {
```

Comment:
It will break `hot` ability, why we need delete it?

Reply:
should we allow setting both hot and hotOnly?

Reply:
won't that apply the plugin twice?

Reply:
hm, I think you are right here, we should not apply plugin twice, but we have logic on webpack-dev-server side (if you have plugin we don't add it again), removing hot is not safe, because we adding client code for HMR when hot is true

Reply:
ah okay then we can remove this